cpp/src/arrow/record_batch.h

```diff
@@ -178,6 +176,8 @@ class ARROW_EXPORT RecordBatchReader {
   /// \return Status
   virtual Status ReadNext(std::shared_ptr<RecordBatch>* batch) = 0;
 
+  Status Next(std::shared_ptr<RecordBatch>* batch) override { return ReadNext(batch); }
```

Comment:
Do you want to open a JIRA about deprecating `ReadNext`?

Reply:
https://issues.apache.org/jira/browse/ARROW-6275